src/ice_shelf/MOM_ice_shelf.F90

```diff
@@ -1497,16 +1681,20 @@ subroutine initialize_ice_shelf(param_file, ocn_grid, Time, CS, diag, forces, fl
   endif
 
   if (new_sim .and. (.not. (CS%override_shelf_movement .and. CS%mass_from_file))) then
-
-    ! This model is initialized internally or from a file.
-    call initialize_ice_thickness(ISS%h_shelf, ISS%area_shelf_h, ISS%hmask, G, US, param_file)
-
+     ! This model is initialized internally or from a file.
+     call initialize_ice_thickness(ISS%h_shelf, ISS%area_shelf_h, ISS%hmask, CS%Grid, CS%Grid_in, US, param_file,&
+          CS%rotate_index, CS%turns)
     ! next make sure mass is consistent with thickness
     do j=G%jsd,G%jed ; do i=G%isd,G%ied
       if ((ISS%hmask(i,j) == 1) .or. (ISS%hmask(i,j) == 2)) then
         ISS%mass_shelf(i,j) = ISS%h_shelf(i,j)*CS%density_ice
       endif
     enddo ; enddo
+    if (CS%debug) then
+       call hchksum(ISS%mass_shelf, "IS init: mass_shelf", G%HI, haloshift=0)
```

Comment:
Please check the units of mass_shelf and area_shelf_h, and add scaling arguments to these `hchksum()` calls as necessary.